### DISCUSSION TITLE
Accessing facts when using JSON as a fact

### DISCUSSION THREAD

**Discussion by Manborough** [hyperjumptech/grule-rule-engine#287] [Q&A]
I've read through the JSON as a fact documentation, and I understand that the original JSON structure will be unchanged from a rules THEN statement, but is there a way to access the JSON structure that's contained in the dataContext that is being altered?

**Comment by newm4n** [comment]
AFAIK, you can access an altered data context (constructed using json) like you would normaly do. The thing is, all alteration only happen in the memory within data context, it will not change the original JSON.

**Comment by codetiger** [comment]
Am using this line to get the resulting fact
`
result := dataContext.Get("jsondata").Value().Interface().(map[string]interface{})
`

**Reply by jonaswouters** [reply]
this works
